### DISCUSSION TITLE
1.42.0

### DISCUSSION THREAD

**Discussion by Koenkk** [Koenkk/zigbee2mqtt#24989] [Releases]
**Update 03-12-2024: in case you lost your paired devices after updating to the 1.42.0-1 Home Assistant add-on, see https://github.com/Koenkk/zigbee2mqtt/discussions/25048**

Feel free to discuss the [1.42.0](https://github.com/Koenkk/zigbee2mqtt/releases/tag/1.42.0) release here!

Also be aware that Zigbee2MQTT 2.0.0 will be released next month, from the changelog:

> # Upcoming Zigbee2MQTT 2.0.0 release
> All the preparations for the 2.0.0 release have now been completed. Note that this release will contain **breaking changes** which can be found [here](https://github.com/Koenkk/zigbee2mqtt/discussions/24198). Starting from 2 December, users can start testing Zigbee2MQTT 2.0.0 by switching to the [dev branch](https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html). If no big blockers are found, 2.0.0 will be released on 3 January 2025.

**Comment by andrehamburg** [comment]
Hi Koenkk, 
i installed 1.42.0-1 but now i'm confused. The instructions says i should move the folder /mnt/data/supervisor/homeassistant/zigbee2mqtt to /mnt/.....   
When i login to Home Assistant by WinSCP the folder /mnt is empty. The folder /zigbee2mqtt is located in the folder /homeassistant. Should i move this folder? And if yes what should be the target folder?
The path in the zigbee2mqtt configuration is /config/zigbee2mqtt. The folder /config is just a link and it is linked to the folder /homeassistant.
Hopefully you can help me.
André

**Reply by vgdh** [reply]
I moved zigbee2mqtt from the config folder to /addon_config and it broke my configuration for good. I have to restore coordinator from the **nvram** backup.

**Comment by ocogneras** [comment]
Hi Andrehambourg,

I'm like you, nothing in /mnt/, and all in /config/homeassistant config is a link through /homeassistant.
if Iook in my zigbeemqtt config file, the path is already "data_path: /config/zigbee2mqtt".

if i have well read update informations, we have to move/copy all in /config/zigbeemqtt in a folder named 486e6e9b_zigbee2mqtt.
I only find this folder in /addon_configs/486e6e9b_zigbee2mqtt, wich is empty too.

Andrehambourg, did you find this folder too? 486e6e9b_zigbee2mqtt in /addon_configs ?
Mine was created the Nov  02, it was the date i installed the 1.40.2-1 ...


Will wate a little if Koenkk or someone else can confirm before doning what i said...

regards.
Olivier

**Reply by ocogneras** [reply]
Tried this:
 - shutdown the module
 - Change the path in configuration : replaced "data_path: / config/zigbee2mqtt" by "data_path: /addon_configs/486e6e9b_zigbee2mqtt"
 - rsync -av /config/zigbee2mqtt/ /addon_configs/486e6e9b_zigbee2mqtt/
 - restart the module
=> Result is KO

Tried this:
- shutdown the module
- Change the path in configuration : replaced "data_path: / config/zigbee2mqtt" by "data_path: /addon_configs/486e6e9b_zigbee2mqtt"
 - rsync -av /config/zigbee2mqtt/ /addon_configs/486e6e9b_zigbee2mqtt/
 - update the module 
 - restart the module
=> Result is KO

Tried this:
- shutdown the module
- Change the path in configuration : replaced "data_path: /addon_configs/486e6e9b_zigbee2mqtt/" by "data_path: / config/zigbee2mqt"
- Module is still updateted
 - restart the module 
 => Result is OK

Even the change is needed for the 2.0 upgrade, or the change is not necessary?

Regards.
Olivier

**Reply by vgdh** [reply]
I have no success with that too.

**Comment by nightfever77** [comment]
There was no 'addon-config**s**' on my setup but 'addon**s**-config'. This was auto-created by another addon. So, a little bit more confusion.
But I managed to get this working:
1. Stopped the addon (1.41...)
2. Created folder '45df7312_zigbee2mqtt' inside config/addons-config
3. Moved zigbee2mqtt from config/ to config/addons-config/45df7312_zigbee2mqtt/
4. Changed the data path to config/addons-config/45df7312_zigbee2mqtt/zigbee2mqtt
5. Started the addon -> Working --> Updated to 1.42 --> Working

**Reply by Loffretto** [reply]
How do move the files from one to another?

**Reply by colin19889** [reply]
My folders were laid out a little differently than nightfever77, but I used the following,

`mv /config/zibee2mqtt /addon_configs/45df7312_zigbee2mqtt`

This put the contents of that original folder in

`/addon_configs/45df7312_zigbee2mqtt/zigbee2mqtt`

**Comment by Hapi-D** [comment]
I had all the issues above. None of the folders were in the location specified. It took me some time to locate the most likely location of these folders. With hindsight I probably can guess the reason for the /mnt/ in the path. Never mind that. 

I found a folder "addon_configs/45df7312_zigbee2mqtt", copied the contents of the "homeassistant/zigbee2mqtt" folder there.
 
The "data path in the addon config" took me some time to understand what that meant and where I might change it. 

"mount addon specific configuration into the addon container"  .. not a clue what that means. I ignored that one and started the addon. Slowly all devices appear as I write this. 

I found the description in the changelog to be somewhat cryptic. I am fully aware that I'm not a home assistant wizard, this process made it very clear once again. In all honesty I have done some therapeutic shouting while going through the procedure. 

Could I politely suggest a guideline for dummies for any similar future change? Thanks in advance.

**Comment by CHoerby** [comment]
What the changlog say: 

> 
    Stop the addon
    Move the /mnt/data/supervisor/homeassistant/zigbee2mqtt folder to /mnt/data/supervisor/addon_configs/45df7312_zigbee2mqtt/zigbee2mqtt (for example using SSH/SMB or the File Explorer/VSCode addon)
    Go to the addon config and change the data_path from /config/zigbee2mqtt to /addon_config/zigbee2mqtt
    Start the addon
<

What I did: 
0. updated to 1.42.0-1
1. stopping the addon
2. moved the folder "config\zigbee2mqtt" to "addon_configs\45df7312_zigbee2mqtt" 
-> before all data/files was in the folder "config\zigbee2mqtt" and now in "addon_configs\45df7312_zigbee2mqtt/**zigbee2mqtt**" 
3. Changed data_path from "/config/zigbee2mqtt" to "/addon_config/zigbee2mqtt" 
  this was only possible via YAML
4. starting the addon
5. waiting, doing nothing, until addon-manger-log appeared - not opening Zigbee2mqtt Web page
-> working

**Reply by vgdh** [reply]
that works, thank you

**Reply by colin19889** [reply]
This worked for me. I was able to do Step 3 in the Configuration tab of the addon. It did take a few minutes for all of my devices/entities to repopulate the first time I brought it up.

**Reply by benalla** [reply]
@CHoerby thanks! That worked for me.

**Comment by z-master42** [comment]
There's a big [Unreleased] above this passage of text. 1.42.0 has nothing about this in the changelog or the release notes.

**Reply by Hapi-D** [reply]
Well yes, actually I noticed that so initially I thought I'd read it later and upgrade to 1.42.0 now... without a closer look to the actual text. However after installing 1.42.0 *all* zigbee devices were gone. Wiped from the internals of HA so to speak. Then I followed the procedure we were discussing here and they magically returned. 
This might be unrelated of course but I don't think so.

**Reply by z-master42** [reply]
I have not changed anything. Z2M config is still in the main directory of HA. I am on 1.42.0 and all devices are still "there".

**Reply by Hapi-D** [reply]
mm.. I believe that is also discussed in the procedure in the bit where it says "automatically migrate". I really read it in full .. too late, but I did.

**Reply by z-master42** [reply]
Yes, maybe. The Z2M addons_config folder has already existed in my system for a month. But it is still empty.

**Comment by Koenkk** [comment]
The issues with the addon should have been fixed now with the 1.42.0-2 version. Make sure to have the config under ` /mnt/data/supervisor/homeassistant/zigbee2mqtt` and use `/config/zigbee2mqtt` as the data path.

**Reply by CHoerby** [reply]
Copying back? 
if i already have moved to "addon_configs\45df7312_zigbee2mqtt" ?

**Reply by i4mr000t** [reply]
I'm also confused now. Do we need to copy it back to config/zigbee2mqtt or addon_configs/45df7312_zigbee2mqtt

**Reply by Hapi-D** [reply]
I really admire brevity, but I think a somewhat clearer guideline would be appreciated. Definitely by me.

**Reply by vgdh** [reply]
> Make sure to have the config under ` /mnt/data/supervisor/homeassistant/zigbee2mqtt`.

This path doesn't exist if user uses simple ssh addon. 
I've used SSH addon and use already existed folder `/addon_configs/45df7312_zigbee2mqtt` to move the config folder `/config/zigbee2mqtt `.

**Reply by mitch66692** [reply]
> The issues with the addon should have been fixed now with the 1.42.0-2 version. Make sure to have the config under ` /mnt/data/supervisor/homeassistant/zigbee2mqtt` and use `/config/zigbee2mqtt` as the data path.

I can confirm it's now working. When updating to 1.42.0-1 i had to restore the backup to 1.41 for missing path; now from 1.41 to 1.42.0-2 is working without any problem.

**Reply by dark-m0de** [reply]
@Koenkk - I'm also a bit confused about your comment. I was upgrading to 1.42.0-2 without problems. And AFTER THAT I migrated the config to the addon_config folder (see https://github.com/Koenkk/zigbee2mqtt/discussions/24989#discussioncomment-11434953 from @CHoerby). I thought I need to do this to be able to use the hass automated backup function.

However, should I now keep the change or revert back?

**Reply by z-master42** [reply]
Leave it as it is now.

**Comment by cristianstoica** [comment]
Subscribing to the comments above. 
@Koenkk 
_Question_: For people running **1.41.0-1** and want to upgrade to **1.42.0-2**, do they need to do **any manual changes**?

**Reply by CHoerby** [reply]
I've made the movement as described above. 
made some backups now ;) 
I updated to -2
made no manual changes
(no reverting of the folder movement)
all worked fine

**Reply by Koenkk** [reply]
> Subscribing to the comments above. 
> 
> @Koenkk 
> 
> _Question_: For people running **1.41.0-1** and want to upgrade to **1.42.0-2**, do they need to do **any manual changes**?

No, 1.41.0-1 to 1.42.0-2 is fine, only people that are already on 1.42.0-1 are affected

**Reply by cristianstoica** [reply]
Just upgraded to 1.42.0-2. Everything went fine. Thanks.

**Comment by Koenkk** [comment]
In case you lost your paired devices after updating to the 1.42.0-1 Home Assistant add-on, see https://github.com/Koenkk/zigbee2mqtt/discussions/25048

**Comment by chris-1243** [comment]
Z2M 1.42.0-dev commit: [71e163fcc3332a61983c12c87dcbca9ab61c266f](https://github.com/Koenkk/zigbee2mqtt/commit/71e163fcc3332a61983c12c87dcbca9ab61c266f)

On a test network, I have this error multiple times ( I did not want to spam Z2M 2.0.0 discussion)
```
[2024-12-04 11:47:22] error:   z2m: EventBus error 'HomeAssistant/publishEntityState': Cannot read properties of null (reading 'toString')
``` 

As dev branch is mainly based on Z2M 2.0.0, `color options: execute_it_off` (IKEA lights) creates a sensor in HA (actually, it is the same on main branch). It reports only the state. Would it be possible to move this option as a light attribute? In my opinion, this sensor is not needed and makes the creation of automation by far more complicated. This is based on my use cases.  For example, it is not possible to get a `true` answer when checking `{{ is_state_attr("light.0xf4b3b1fffe3ff501", "color_options", {"execute_if_off": True}) }}` in HA.

![Untitled](https://github.com/user-attachments/assets/55988ac2-b36d-4922-9a22-34daa309f168)

**Reply by Koenkk** [reply]
Could you provide the debug log of this?

See [this](https://www.zigbee2mqtt.io/guide/configuration/logging.html) on how to enable debug logging.

**Reply by chris-1243** [reply]
The debug log as requested

[log.txt](https://github.com/user-attachments/files/18019273/log.txt)

**Reply by abrasher** [reply]
I also have this error in #25101

**Comment by chris-1243** [comment]
Any change how the permit join button should work? On standard 1.42.0, a click on it allow permit join and a second click stop it. This is expected behavior.
On 1.42.0-dev commit: [639e3bb6fd50928da002d417455b5eaf3acb2ad8](https://github.com/Koenkk/zigbee2mqtt/commit/639e3bb6fd50928da002d417455b5eaf3acb2ad8), a click on it allow permit join and a second click just reset the countdown.

By the way, the homeassistant log is full of such warnings:

```
homeassistant  | 2024-12-06 09:34:08.830 WARNING (MainThread) [homeassistant.helpers.template] Template variable warning: 'dict object' has no attribute 'permit_join' when rendering '{{ value_json.permit_join | lower }}'
``` 

Z2M debug log [log.txt](https://github.com/user-attachments/files/18035114/log.txt)
homeassistant log 
[home-assistant.log](https://github.com/user-attachments/files/18035135/home-assistant.log)

**Reply by Koenkk** [reply]
Should be fixed in the dev branch (z2m 2.0.0)

**Reply by chris-1243** [reply]
I just had a try and perfectly working

Thanks for the fix

**Comment by starox** [comment]
Just a tip for those which have a lot of device on a wide geo area and sometime not very practical ...
Get a zigbee button and program it to activate permit join.
This is friendly ;-)

**Comment by Burki24** [comment]
@Koenkk, @Nerivec ,
Short question regarding handling of groups:

until 1.42.x-dev it was possible to get a list of all groups from zigbee2mqtt. Since then, I get no response with the following tries:

- bridge/request/group/list
- this.zigbee.getGroups
- this.zigbee.controller?.getGroups

This runs before without any problems:
```
        if (data.topic.startsWith(`${this.baseTopic}/SymconExtension/request/getGroupInfo`)) {
            try {
                const groupname = data.topic.split('/').slice(4).join('/');
                const message = JSON.parse(data.message);
                const groupExposes = this.#createGroupExposes(groupname);
                groupExposes.transaction = message.transaction;
                this.logger.info('Symcon: request/getGroupe');
                await this.mqtt.publish(`SymconExtension/response/getGroupInfo/${groupname}`, JSON.stringify(groupExposes), { retain: false, qos: 0 }, `${this.baseTopic}`, false, false);
            } catch (error) {
                let errormessage = 'Unknown Error'
                if (error instanceof Error) errormessage = error.message
                this.logger.error(`Symcon error (${errormessage}) at Topic ${data.topic}`);
            }
            return;
        }
        if (data.topic == `${this.baseTopic}/SymconExtension/lists/request/getGroups`) {
            try {
                const message = JSON.parse(data.message);
                const groups = {
                    list: [],
                    transaction: 0,
                };
                groups.list = this.settings.getGroups();
                groups.transaction = message.transaction;
                this.logger.info('Symcon: lists/request/getGroups');
                await this.mqtt.publish('SymconExtension/lists/response/getGroups', JSON.stringify(groups), { retain: false, qos: 0 }, `${this.baseTopic}`, false, false);
            } catch (error) {
                let errormessage = 'Unknown Error'
                if (error instanceof Error) errormessage = error.message
                this.logger.error(`Symcon error (${errormessage}) at Topic ${data.topic}`);
            }
            return;
        }
```

Is there any other solution for that?

Thanks in advance

Burkhard

**Reply by Nerivec** [reply]
> bridge/request/group/list
> this.zigbee.getGroups
> this.zigbee.controller?.getGroups

I don't think any of these ever existed.

I'm not sure what that first `if` is doing, but in the second one, it uses settings to get the groups (`this.settings.getGroups()`), so, that should still work, but that wrapper function was removed (no longer used), so you need to get the groups directly `this.settings.get().groups` and map whatever properties you need.
See old implementation:
https://github.com/Koenkk/zigbee2mqtt/blob/bc6eccf6152c4cd91e75309f7ea0b8ad4e8835d4/lib/util/settings.ts#L573-L579
You also have this iterator depending on needs:
https://github.com/Koenkk/zigbee2mqtt/blob/47d73eb23a3c452f1757587184ba5a066f0e5edc/lib/zigbee.ts#L346

I'm assuming this is an external extension, did you migrate the constructor properly too?
https://github.com/Koenkk/zigbee2mqtt.io/blob/dev/docs/advanced/more/external_extensions.md#external-extensions
